### PR TITLE
fix(testing): `assertObjectMatch` matches ArrayBuffer views correctly

### LINKED ISSUE #1842: `assertObjectMatch` does not correctly match `Uint8Array`
**Describe the bug**

When comparing `Uint8Array` values within an object, `assertObjectMatch` throws an `AssertionError` claiming that the `Uint8Array` fields are not equal.

**Steps to Reproduce**

Run the following test
```js
import { assertObjectMatch } from "https://deno.land/std@0.122.0/testing/asserts.ts";

Deno.test("should pass comparing Uint8Arrays", async () => {
  const body = new Uint8Array();

  function foo() {
    return { body, foo: "bar", fizz: 'buzz' };
  }

  assertObjectMatch(foo(), { body, foo: 'bar' });
});
```

which should log, failing the test:
```sh
    [Diff] Actual / Expected


    {
+     body: Uint8Array(0) [
+       ,
+     ],
-     body: {},
      foo: "bar",
    }

    at assertEquals (https://deno.land/std@0.122.0/testing/asserts.ts:269:9)
    at assertObjectMatch (https://deno.land/std@0.122.0/testing/asserts.ts:591:10)
    ...
```

**Expected behavior**

The assertion to pass successfully, since the expected value is a subset object of the actual object

**Environment**

- OS: macOS 11.2.3
- deno version: 1.18.0
- std version: 0.122.0

### PR DESCRIPTION
**Pull request by ah-yu**
Closes #1842

### CODE REVIEW

File: testing/asserts_test.ts [hunk 1]
```diff
@@ -601,6 +603,8 @@ Deno.test("testingAssertObjectMatching", function (): void {
     assertObjectMatch({ test: { a: 1 } }, { test: new A(1) });
     assertObjectMatch({ test: new A(1) }, { test: new A(1) });
   }
+  // subset with TypedArray instance inside
+  assertObjectMatch(m, { bar: new Uint8Array([1, 2]) });
```
Comment (kt3k) [inline testing/asserts_test.ts:607]: The fix looks good to me, but this test case seems passing even without the change.

If I changed the assertion to `assertObjectMatch(m, m)`, then the issue described in #1842 reproduces as a test case. So maybe let's add that assertion here?

**Review by kt3k — COMMENTED** [review]
@ah-yu Thanks for the PR. Left one comment

**Comment by ah-yu** [comment]
@kt3k Thanks for review! I think I made a bad fix before.
This bug caused by using same `weakmap` between `actual` computes and `expected` computes.
```js
const foo = new Map(); // Array/TypedArray/etc will cause the same problem
assertObjectMatch({ foo, bar:"bar"},{foo}); // fails 
```

File: testing/asserts.ts [hunk 2]
```diff
@@ -540,53 +540,51 @@ export function assertObjectMatch(
   expected: Record<PropertyKey, unknown>,
 ): void {
   type loose = Record<PropertyKey, unknown>;
-  const seen = new WeakMap();
-  function filter(a: loose, b: loose): loose {
-    // If the actual value is an array, let assertEquals do the assertion.
-    if (Array.isArray(a)) {
```
Comment (ah-yu) [inline testing/asserts.ts:546]: Maybe we could remove this?

**Comment by ah-yu** [comment]
Also fix https://github.com/denoland/deno_std/issues/1288 
Seems that https://github.com/denoland/deno_std/pull/1289 is not a right fix.

**Comment by ah-yu** [comment]
Fixes https://github.com/denoland/deno_std/pull/1289#issuecomment-922692148

**Review by kt3k — APPROVED** [review]
@ah-yu LGTM. Thanks!

**Pull request merged**